hw/nfc-nci.c

```diff
@@ -662,28 +663,20 @@ nfc_delivery_deactivate_cmd_cb(void* data, union nci_packet* pkt)
     assert(data);
 
     param = (struct nfc_deactivate_ntf_param*)data;
-    res = nfc_create_deactivate_ntf(param->type, param->reason, pkt);
+    res = nfc_create_deactivate_ntf(nfc, param->type, param->reason, pkt);
 
     qemu_free(param);
 
     return res;
 }
 
-static size_t
-init_process_oid_rf_deactivate_cmd(const union nci_packet* cmd,
-                                   struct nfc_device* nfc,
-                                   union nci_packet* rsp,
-                                   struct nfc_delivery_cb* cb)
+static void
+nfc_rf_state_deactivate_transition(struct nci_rf_deactivate_cmd* payload,
+                                   struct nfc_device* nfc)
```

Comment:
I'm not yet sure if this function is the best way to handle the problem, but the first parameter should not by the payload. Please pass the relevant fields (e.g, 'type') directly.